packages/shared/src/graphql/squads.ts

```diff
@@ -303,3 +349,26 @@ export async function editSquad(
   );
   return data.editSquad;
 }
+
+export const verifyPermission = (
+  squad: Squad,
+  permission: SourcePermissions,
+): boolean => {
+  const hasPermission =
+    !!squad?.currentMember?.permissions?.includes(permission);
+
+  // TODO WT-1224 find better way to do this
+  if (hasPermission && permission === SourcePermissions.Post) {
+    const membersCanPost = squad.memberPostingRole === SourceMemberRole.Member;
+
+    if (membersCanPost) {
+      return true;
+    }
+
+    return [SourceMemberRole.Owner, SourceMemberRole.Moderator].includes(
+      squad.currentMember.role,
+    );
+  }
+
+  return hasPermission;
```

Comment:
@rebelchris so what I want to do is to check `memberPostingRole` in addition to `SourcePermissions.Post` permission. Problem is this will need to be updated if we add more permission options for users in squad settings eg. only owners can post. 

Now it works because by default everyone can post and you can only set moderator or above in squad creation/settings.

I was thinking maybe we can expose array of roles which would then be used for the check. It would be simple filter on the API side which would then only return roles that are higher then `memberPostingRank` and we could use that instead of hardcoded array above to check.

Other thing is that I might have been wrong and exposing `squad.memberRoleRank` and `currentMember.roleRank` (numeric values) to the frontend would be useful and then we can just check if current rank value is higher then the one set for a squad. 

WDYT?

In hindsight it would be awesome if we could somehow just omit the `SourcePermissions.Post` permission form `currentMember.permissions` depending on the `sourceMemberRole` squad setting and current user on the API side. Not sure how performant that query would be.

Reply:
I agree the later is actually best case.
Dependant on squad `memberPostingRank` we evaluate if permissions should include `Post`.

Happy to include that change as it makes things more explicit and clear, let's do that?

Reply:
@rebelchris so the last one on the API side?

Reply:
Implementation here https://github.com/dailydotdev/daily-api/pull/1232